src/main/java/edu/harvard/dbmi/avillach/dictionaryetl/facet/FacetLoaderService.java

```diff
@@ -0,0 +1,429 @@
+    private static final String KEY_FACET_EXPRESSION_GROUPS_HASH = "facet_loader.expression_groups_sha256hex";
+
+    public static final String KEY_EFFECTIVE_EXPRESSION_GROUPS = "facet_loader.effective_expression_groups";
+    private static final String KEY_EFFECTIVE_EXPRESSION_GROUPS_HASH = "facet_loader.effective_expression_groups_sha256hex";
+
+    public FacetLoaderService(FacetCategoryRepository facetCategoryRepository, FacetRepository facetRepository,
+                              ConceptRepository conceptRepository, FacetConceptRepository facetConceptRepository,
+                              FacetMetadataRepository facetMetadataRepository,
+                              ObjectMapper objectMapper) {
+        this.facetCategoryRepository = facetCategoryRepository;
+        this.facetRepository = facetRepository;
+        this.conceptRepository = conceptRepository;
+        this.facetConceptRepository = facetConceptRepository;
+        this.facetMetadataRepository = facetMetadataRepository;
+        this.objectMapper = objectMapper.copy().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
+    }
+
+    @Transactional
+    public ClearResult clear(FacetClearRequest request) {
+        long categoriesDeleted = 0;
+        long facetsDeleted = 0;
+        long mappingsDeleted = 0;
+
+        if (request == null) return new ClearResult(0,0,0);
+
+        // 1) Clear by category names
+        if (request.facetCategories != null) {
+            for (String catName : request.facetCategories) {
+                if (catName == null || catName.isBlank()) continue;
+                Optional<FacetCategoryModel> opt = facetCategoryRepository.findByName(catName);
+                if (opt.isEmpty()) continue;
+                FacetCategoryModel cat = opt.get();
+
+                // count before delete
+                long facetCount = facetRepository.countByFacetCategoryId(cat.getFacetCategoryId());
+                long mappingCount = facetConceptRepository.countAllForCategory(cat.getFacetCategoryId());
+
+                // delete mappings and facets for category
+                facetConceptRepository.deleteAllForCategory(cat.getFacetCategoryId());
+                facetRepository.deleteAllForCategory(cat.getFacetCategoryId());
+
+                // delete category
+                facetCategoryRepository.deleteById(cat.getFacetCategoryId());
+
+                categoriesDeleted += 1;
+                facetsDeleted += facetCount;
+                mappingsDeleted += mappingCount;
+            }
+        }
+
+        // 2) Clear by facet names (including their descendants)
+        if (request.facets != null) {
+            for (String facetName : request.facets) {
+                if (facetName == null || facetName.isBlank()) continue;
+                Optional<FacetModel> optFacet = facetRepository.findByName(facetName);
+                if (optFacet.isEmpty()) continue;
```

Comment:
same comment here about logging

Reply:
added logging statement to inform user the facet wasn't found with given facet name